profilehooks.py

```diff
@@ -777,43 +787,47 @@ def __init__(
         self.totaltime = 0
         self.immediate = immediate
         self.timer = timer
+        self.enable = enable
         if not immediate:
             atexit.register(self.atexit)
 
     def __call__(self, *args, **kw):
         """Profile a singe call to the function."""
         fn = self.fn
+        if not self.enable:
+            return fn(*args, **kw)
         timer = self.timer
         self.ncalls += 1
         start = timer()
         try:
             return fn(*args, **kw)
         finally:
-            duration = timer() - start
-            self.totaltime += duration
-            if self.immediate:
-                funcname = fn.__name__
-                filename = fn.__code__.co_filename
-                lineno = fn.__code__.co_firstlineno
-                message = "%s (%s:%s):\n    %.3f seconds\n\n" % (
-                    funcname, filename, lineno, duration,
-                )
-                if self.logger:
-                    self.logger.log(self.log_level, message)
-                else:
-                    sys.stderr.write("\n  " + message)
-                    sys.stderr.flush()
+            if self.enable:
+                duration = timer() - start
+                self.totaltime += duration
+                if self.immediate:
+                    funcname = fn.__name__
+                    filename = fn.__code__.co_filename
+                    lineno = fn.__code__.co_firstlineno
+                    message = "%s (%s:%s):\n    %.3f seconds\n\n" % (
+                        funcname, filename, lineno, duration,
+                    )
+                    if self.logger:
+                        self.logger.log(self.log_level, message)
+                    else:
+                        sys.stderr.write("\n  " + message)
+                        sys.stderr.flush()
 
     def atexit(self):
-        if not self.ncalls:
+        if not self.ncalls and not self.enable:
```

Comment:
I think this should be` if not self.ncalls or not self.enable`.

(But actually if self.enable is False then self.ncalls will always be 0, won't it?)